packages/contracts/src/core/dao/DAO.sol

```diff
@@ -122,6 +154,28 @@ contract DAO is
         __PermissionManager_init(_initialOwner);
     }
 
+    /// @notice Initializes the DAO after an upgrade from a previous version.
+    /// @param _protocolVersion The protocol version number of the previous DAO implementation contract this upgrade is transitioning from.
+    /// @param _initData The initialization data to be passed to via `upgradeToAndCall` (see [ERC-1967](https://docs.openzeppelin.com/contracts/4.x/api/proxy#ERC1967Upgrade)).
+    function initializeFrom(
+        uint8[3] calldata _protocolVersion,
+        bytes calldata _initData
+    ) external reinitializer(2) {
+        _initData; // Silences the unused function parameter warning.
+
+        // Check that the contract is not upgrading from a different major release.
+        if (_protocolVersion[0] != 1) {
```

Comment:
Doesn't the PSP already prevent upgrades across releases?

Reply:
We don't use the PSP to upgrade the DAO. The SDK has to create a proposal in a governance plugin.

Reply:
When you prepare the update, don't you register it on the PSP?

Reply:
No, the `PluginSetupProcessor` consumes `PluginSetup`s contracts from a `PluginRepo`. It expects `prepareInstallation`, and `prepareUninstallation` function to be inside.

Reply:
Totally, huge lapsus 🤦